racket-cas/racket-cas.rkt

```diff
@@ -2851,7 +3040,8 @@
   (define (extract u)
     (match u
       [(list 'or e ...) (map solution e)]
-      [(list _ '= x0)   (list x0)]
+      [(list '= y x0) #:when (equal? y x)
+                      (list x0)]
       [_                '()]))
```

Comment:
Nice catch. Include this in the solve PR.
Is there a test that can catch these problems?